source/Uthernet2.cpp

```diff
@@ -912,7 +912,7 @@ void Uthernet2::resetRXTXBuffers(const size_t i)
 void Uthernet2::openSystemSocket(const size_t i, const int type, const int protocol, const int status)
 {
     Socket &s = mySockets[i];
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__APPLE__)
```

Comment:
When is the socket set as non-blocking then?

Someone asked me once to add this

https://github.com/audetto/AppleWin/blob/0338d8fd1baf60eefa31ceffe09ae20d3d6e9087/source/Uthernet2.cpp#L81

If it is not good for a MacOS we can remove it

Reply:
`SOCK_NONBLOCK` is a Linuxism and doesn't exist in BSD/macOS AFAIK. According to the manpage:

> An fcntl(2) call can be used to [...] enable non-blocking I/O

so I'll investigate that separately. Either way, `SOCK_NONBLOCK` is a build error in macOS, and using the `#ifdef` hack causes it to fail with "Protocol wrong type for socket" and [hang](https://github.com/AppleWin/AppleWin/issues/1088#issuecomment-1193389243). Do you know if whoever suggested that hack actually test the fix or was just working around the build error?

Reply:
Okay, so I think this is the right fix:

```diff
diff --git a/source/Uthernet2.cpp b/source/Uthernet2.cpp
index 461c0a72..870c17bd 100644
--- a/source/Uthernet2.cpp
+++ b/source/Uthernet2.cpp
@@ -77,12 +77,8 @@ typedef int socklen_t;
 
 #endif
 
-// fix SOCK_NONBLOCK for e.g. macOS
-#ifndef SOCK_NONBLOCK
-// DISCALIMER
-// totally untested, use at your own risk
+#ifdef __APPLE__
 #include <fcntl.h>
-#define SOCK_NONBLOCK O_NONBLOCK
 #endif
 
 // Dest MAC + Source MAC + Ether Type
@@ -930,6 +926,8 @@ void Uthernet2::openSystemSocket(const size_t i, const int type, const int proto
 #ifdef _MSC_VER
         u_long on = 1;
         ioctlsocket(fd, FIONBIO, &on);
+#elif __APPLE__
+        fcntl(fd, O_NONBLOCK);
 #endif
         s.setFD(fd, status);
     }
```

and both ip65 and a2stream seem to work fine. Before I push, do you know of anything I should specifically test? (I'm not sure how the blocking or non-blocking manifests.)

There might be a more generic BSD Unix `#ifdef` but I think I'll leave that to somebody who could actually test it.